mlflow/pyfunc/backend.py

```diff
@@ -282,23 +286,66 @@ def can_score_model(self):
             # Can not find conda
             return False
 
+    def generate_dockerfile(
+            self, model_uri, output_directory="generate_dockerfile_output", install_mlflow=False, mlflow_home=None, enable_mlserver=False
+    ):
+        copy_model_into_container = self.copy_model_into_container_wrapper(model_uri, install_mlflow, enable_mlserver)
+        pyfunc_entrypoint = _pyfunc_entrypoint(self._env_manager, model_uri, install_mlflow, enable_mlserver)
+
+        mlflow_home = os.path.abspath(mlflow_home) if mlflow_home else None
+
+        is_conda = self._env_manager == em.CONDA
+        setup_miniconda = SETUP_MINICONDA if is_conda else ""
+        setup_pyenv_and_virtualenv = "" if is_conda else SETUP_PYENV_AND_VIRTUALENV
+
+        os.makedirs(output_directory, exist_ok=True)
+
+        _logger.debug("Created all folders in path", extra={"output_directory": output_directory})
+        install_mlflow = _get_mlflow_install_step(output_directory, mlflow_home)
+
+        custom_setup_steps = copy_model_into_container(output_directory) if copy_model_into_container else ""
+        dockerfile_text = _generate_dockerfile_raw(
+            setup_miniconda=setup_miniconda,
+            setup_pyenv_and_virtualenv=setup_pyenv_and_virtualenv,
+            install_mlflow=install_mlflow,
+            custom_setup_steps=custom_setup_steps,
+            entrypoint=pyfunc_entrypoint
+        )
+        _logger.debug("generated dockerfile text", extra={"dockerfile": dockerfile_text})
+
+        with open(os.path.join(output_directory, "Dockerfile"), "w") as dockerfile:
+            dockerfile.write(dockerfile_text)
+
     def build_image(
```

Comment:
While we're at it, can we add the `build_image` and `generate_dockerfile` base methods to the FlavorBackend base class (https://github.com/mlflow/mlflow/blob/9ea22cdf4d18204e6b92585a028d40341667ac6e/mlflow/models/flavor_backend.py#L4) with a default implementation of `raise NotImplementedError`?

Reply:
Done